java-test-samples/apigw-lambda-list-s3-buckets/src/test/java/com/example/AppIntegrationTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT-0
+ */
+
+package com.example;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+
+@Testcontainers
+public class AppIntegrationTest {
+  private static S3Client s3Client;
+
+  private static final DockerImageName localStackImage =
+    DockerImageName.parse("localstack/localstack:0.14.3");
+
+  @Rule
+  public static final LocalStackContainer localstack = new LocalStackContainer(localStackImage).withServices(S3);
+
+  @BeforeAll
+  static void setup() {
+    localstack.start();
+
+    s3Client = S3Client
+      .builder()
+      .endpointOverride(localstack.getEndpointOverride(LocalStackContainer.Service.S3))
+      .credentialsProvider(
+        StaticCredentialsProvider.create(
+          AwsBasicCredentials.create(localstack.getAccessKey(), localstack.getSecretKey())
+        )
+      )
+      .region(Region.of(localstack.getRegion()))
+      .build();
+
+    s3Client.createBucket(b -> b.bucket("foo"));
+    s3Client.createBucket(b -> b.bucket("bar"));
+  }
+
+  @AfterAll
+  static void cleanup() {
+    localstack.stop();
+  }
+
+  @Test
+  public void successfulResponse() {
+    App app = new App(s3Client);
+    APIGatewayProxyResponseEvent result = app.handleRequest(null, null);
```

Comment:
Not realistic (use events and lambda-test library)

Reply:
Fixed